static/specification.html

```diff
@@ -138,605 +335,536 @@ <h2>
         <li>Developers have a way to associate payments with their users, in
         order to provide a range of experiences.
         </li>
-        <li>Pays continuously as the user consumes content.
+        <li>Users retains control of if, when, and how payments are made to the
+        site. For example, micropayments of a predetermined monetary value
+        could be "streamed" to the site over time. Alternatively, the user
+        could make one or many discreet "one-off" payment(s) of any arbitrary
+        monetary amount, even while micropayments are simultaneously being
+        streamed.
         </li>
       </ul>
     </section>
     <section>
       <h2>
-        Extensions to the `Document` interface
+        Link type "monetization"
       </h2>
-      <pre class="idl">
-        partial interface Document {
-          readonly attribute Monetization monetization;
-        };
-      </pre>
+      <aside class="issue">
+        <p>
+          This is a "monkey patch" for the [[HTML]] specification: If accepted,
+          this will eventually be moved to HTML itself.
+        </p>
+      </aside>
+      <aside class="note" title="Multiple monetization links">
+        <p>
+          Although a single HTML document can have multiple [^link^] elements
+          with the "monetization" link type, the user agent might restrict the
+          number of concurrent [=payment connections=] it can simultaneously
+          support. In such a case, an "error" event will be fired at the
+          [^link^] element where a connection can't be established.
```

Comment:
Without any real detail on the error event how would a web application frontend distinguish this (connection limit exceeded) from an SPSP error?

Reply:
I guess the question (for those more knowledgeable in this space) is if developers can do anything meaningful with the distinction? We couldn’t originally come up with anything so I’m interested in hearing use cases.

Reply:
Well let's start by listing the current errors:
1. SPSP request failed
2. SPSP JSON malformed
3. Connection limit exceeded 
4. ?  What else am I missing

I guess I'm wondering about user feedback for 3. What would it look like to them? 
It's just "not working"

Reply:
Continuing on with this. There was a time in the evolution of the spec where the "error" event bubbled up the tree, but then it occurred to author that that would be problematic as there would be an unintended consequence of it triggering peoples handlers for window etc. 

There was also some discussion as to whether ErrorEvent might be more suitable as it had an `error` field where more data would be attached. It was found not to be suitable. 

We then went back to simply Event('error'). 

ppk suggested we at the least log error information to the console when localStorage.WM_DEBUG was set 

However, it's possible we could use a MonetizationError Event subclass with type 'monetizationerror' which we could actually endow with some useful information.  It could also bubble up the tree like the monetization events.

Reply:
@AlexLakatos 

Can you speak to this, given you've effectively inherited the PR ?

The overview comment includes:
- Developer-recoverable errors are handled via the <link onerror> event handler.

The spec itself talks nothing of what errors are classed as developer recoverable, and which are non-recoverable. 
My recollection is quite fuzzy, but I gathered from somewhere it means that the payment pointer can possibly be changed to remedy the 'error' state ? 

> Without any real detail on the error event how would a web application frontend distinguish this (connection limit exceeded) from an SPSP error?

^

Reply:
The only action for every error that's "recoverable" from a developer's point of view is to change the payment pointer on the page. 

We thought adding extra information to the error won't help in this case, and a custom event, or even ErrorEvent wouldn't help here. A simple Error event would be better here.

Reply:
https://github.com/WICG/webmonetization/pull/193#pullrequestreview-1215158725